ci/cccp_ci.py

```diff
@@ -43,36 +66,68 @@ def get_nodes(ver="7", arch="x86_64", count=4):
 
 
 def print_nodes():
+    """
+    Function to print nodes from a local file env.properties.
+    """
     with open('env.properties') as f:
         s = f.read()
 
     _print('\n'.join(s.splitlines()[3:]))
 
 
+def _if_debug():
+    """
+    If whitelisted github user has added "dotests-debug" string as comment
+    on a given PR, the nodes are kept for 2 hours for debugging, once
+    time lapses the nodes are returned to CICO infrastructure.
+    """
+    if DEBUG:
+        _print('Reserving nodes for debugging...')
+        _print('=' * 10 + 'Node Info' + '=' * 10)
+        print_nodes()
+        try:
+            _print('Sleeping for %s seconds for debugging...'
+                   % 7200)
+            import time
+            time.sleep(int(7200))
+        except Exception as e:
+            _print(e)
+        with open('env.properties', 'a') as f:
+            f.write('\nBUILD_FAILED=true\n')
+    sys.exit(1)
+
+
 if __name__ == '__main__':
     try:
+        # get nodes from CICO infra
         nodes = get_nodes(count=5)
+        _print(str(nodes))
+    except Exception as e:
+        _print('Build failed while receiving nodes from CICO: %s' % e)
```

Comment:
Can we change the message to something like `could not get node from CICO for setup`. Also I would say lets add one more iteration for getting nodes. if it fails second time, we exit. Because this has got nothing to do with the service code and if we dont get node, CI is going to fail unnecessarily.

Reply:
Noted. Will have a separate PR handling this.